### PR TITLE
Jimbrooke/structure

### PR DESCRIPTION
**Pull request by jimbrooke**
This contains the basic structure of C++ classes which represent the firmware.  Many classes currently do nothing, but they will ultimately contain the code which controls the firmware.

Changes wrt Joeal's initial PR :
1. add namespaces
2. remove unnecessary #includes
3. remove DTPNode
4. rename/move test apps
5. get connections file (dtp_connections.xml) from $DUNEDAQ_SHARE_PATH
6. add top-level DTPPodNode
7. add full set of node classes (inc several placeholders)
8. fix style issues (snake case)

### CODE REVIEW

**Comment by jimbrooke** [comment]
Class structure is there, with a basic test that it works at test/apps/test_nodes.cxx

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+#emacs temp files
+
+*~
+*\#
+\.\#*
+
+# clang and LSP files
+
+.cache/*
```
Comment (alessandrothea) [inline .gitignore]: Why so many entries? Should really not be required when working in a DUNE-DAQ area.
Possibly a sign of development environment abuse!

File: include/dtpcontrols/DataReceptionNode.hpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+class DataReceptionNode : public uhal::Node {
+
+  UHAL_DERIVEDNODE(DataReceptionNode)
+  
+public:
+  explicit DataReceptionNode(const uhal::Node& );
+  virtual ~DataReceptionNode();
+
+  void reset_input_word_counter(bool);
```
Comment (alessandrothea) [inline include/dtpcontrols/DataReceptionNode.hpp]: I suggest setting a default for the dispatch flag (the name of which should appear in the header, not to leave the meaning of the boolean argument to the imagination of the reader).
I would set the default to `true`, as the actions encoded in methods are typically immediately executed.

**Review by alessandrothea — COMMENTED** [review]
Looks good!

**Comment by alessandrothea** [comment]
I suspect that the pipeline failure is due to a broken cmakeConfig in `uhallibs`. I think I've fixed that and pushed it to `develop`. If I'm right the failing pipeline will work tomorrow after the nightly has completed

**Comment by alessandrothea** [comment]
Commit in question
https://github.com/DUNE-DAQ/uhallibs/commit/23f006d689ff23cfebfaa24268394ad3ea484dba

**Comment by jimbrooke** [comment]
thanks @alessandrothea.  if that works I think this PR is ready for merging - unless you see any issues.  i've fixed the ones you raised above.

**Review by alessandrothea — APPROVED** [review]

**Pull request merged**